crate/server/src/database/redis/objects_db.rs

```diff
@@ -108,22 +116,31 @@ impl ObjectsDB {
     }
 
     fn encrypt_object(&self, uid: &str, redis_db_object: &RedisDbObject) -> KResult<Vec<u8>> {
-        let mut nonce_bytes = [0; Aes256Gcm::NONCE_LENGTH];
-        kdf256!(&mut nonce_bytes, uid.as_bytes());
-        let ciphertext = self.dem.encrypt(
-            &Nonce::from(nonce_bytes),
+        let nonce = {
+            let mut rng = self.rng.lock().expect("failed acquiring a lock on the RNG");
+            Nonce::new(&mut *rng)
+        };
+        let mut ciphertext = nonce.as_bytes().to_vec();
+        ciphertext.extend(self.dem.encrypt(
+            &nonce,
             &serde_json::to_vec(redis_db_object)?,
-            None,
-        )?;
+            Some(uid.as_bytes()),
+        )?);
         Ok(ciphertext)
     }
 
     fn decrypt_object(&self, uid: &str, ciphertext: &[u8]) -> KResult<RedisDbObject> {
-        let mut nonce_bytes = [0; Aes256Gcm::NONCE_LENGTH];
-        kdf256!(&mut nonce_bytes, uid.as_bytes());
-        let plaintext = self
-            .dem
-            .decrypt(&Nonce::from(nonce_bytes), ciphertext, None)?;
+        if ciphertext.len() <= Aes256Gcm::NONCE_LENGTH {
+            return Err(KmsError::CryptographicError(
+                "invalid ciphertext".to_string(),
+            ))
+        }
```

Comment:
You can use the `ENCRYPTION_OVERHEAD` constant if it still exists.
```suggestion
        if ciphertext.len() <= Aes256Gcm::NONCE_LENGTH {
            return Err(KmsError::CryptographicError(
                "invalid ciphertext".to_string(),
            ))
        }
```

Reply:
Done